examples/iterative_sense_reconstruction.py

```diff
@@ -0,0 +1,140 @@
+#
+# $ y = Ax + n $
+#
+# where $n$ describes complex Gaussian noise. The image $x$ can be obtained by minimising the functionl $F$
+#
+# $ x = \argmin_x F(x) $
+#
+# $ F(x) = \frac{1}{2}||W^{\frac{1}{2}}(Ax - y)||_2^2 $
+#
+# where $W^\frac{1}{2}$ is the square root of the density compensation function (which corresponds to a diagonal
+# operator).
+#
+# Setting the derivative of the functional $F$ to zero and rearranging yields
+#
+# $ A^H W A x = A^H W y$
+#
+# which is a linear system $Hx = b$ that needs to be solved for $x$.
+# %%
+import mrpro
+
+# Use the trajectory that is stored in the ISMRMRD file
+trajectory = mrpro.data.traj_calculators.KTrajectoryIsmrmrd()
+# Load in the Data from the ISMRMRD file
+kdata = mrpro.data.KData.from_file(data_file.name, trajectory)
+kdata.header.recon_matrix.x = 256
+kdata.header.recon_matrix.y = 256
+
+# %%
+iterative_sense_reconstruction = mrpro.algorithms.reconstruction.IterativeSenseReconstruction.from_kdata(
+    kdata, n_iterations=4
+)
+img = iterative_sense_reconstruction(kdata)
+
+
+# %% [markdown]
+# ### Behind the scenes
+
+# %% [markdown]
+# ##### $W$
+
+# %%
+# The density compensation operator is calculated based on the k-space locations of the acquired data.
+dcf_operator = mrpro.data.DcfData.from_traj_voronoi(kdata.traj).as_operator()
+
+
+# %% [markdown]
+# ##### $A$
+
+# %%
+# Define Fourier operator using the trajectory and header information in kdata
+fourier_operator = mrpro.operators.FourierOp.from_kdata(kdata)
+
+# Calculate coil maps
+# Note that operators return a tuple of tensors, so we need to unpack it,
+# even though there is only one tensor returned from adjoint operator.
+img_coilwise = mrpro.data.IData.from_tensor_and_kheader(*fourier_operator.H(*dcf_operator(kdata.data)), kdata.header)
+csm_operator = mrpro.data.CsmData.from_idata_walsh(img_coilwise).as_operator()
+
+# Create the acquisition operator A
+acquisition_operator = fourier_operator @ csm_operator
+
+# %% [markdown]
+# ##### $b = A^H W y$
+
+# %%
+(right_hand_side,) = acquisition_operator.H(dcf_operator(kdata.data)[0])
+
+
+# %% [markdown]
+# ##### $H = A^H W A$
+
+# %%
+operator = acquisition_operator.H @ dcf_operator @ acquisition_operator
+
+# %% [markdown]
+# ##### Run conjugate gradient
+
+# %%
+import torch
+
+img_manual = mrpro.algorithms.optimizers.cg(
+    operator, right_hand_side, initial_value=right_hand_side, max_iterations=4, tolerance=0.0
+)
+
+
+# %%
+# For comparison we can also carry out a direct reconstruction
+direct_reconstruction = mrpro.algorithms.reconstruction.DirectReconstruction.from_kdata(kdata)
+img_direct = direct_reconstruction(kdata).rss().cpu()
+img_direct = img_direct
+
+# %%
+# Display the reconstructed image
+import matplotlib.pyplot as plt
+
+fig, ax = plt.subplots(1, 3, squeeze=False)
+ax[0, 0].imshow(img_direct.data[0, 0, :, :])
+ax[0, 0].set_title('Direct Reconstruction', fontsize=10)
+ax[0, 1].imshow(torch.abs(img.data[0, 0, 0, :, :]))
```

Comment:
rss returns a magnitude tensor.

img_direct is a tensor in this example, but img is an IData.
this should be the same in both cases 
consider moving both RSS calls to here, and do  imshow(img.rss[0, 0, 0, :, :])) and imshow(img_direct.rss[0, 0, 0, :, :]))
(remove the RSS above)

Reply:
wouldn't imshow(img.rss[0, 0, 0, :, :])) be confusing because it would suggest that img is a coil-resolved image (same actually applies to img_direct)? Would 

```
img_direct = direct_reconstruction(kdata)

# %%
# Display the reconstructed image
import matplotlib.pyplot as plt

fig, ax = plt.subplots(1, 3, squeeze=False)
ax[0, 0].imshow(torch.abs(img_direct.data[0, 0, :, :]))
ax[0, 0].set_title('Direct Reconstruction', fontsize=10)
ax[0, 1].imshow(torch.abs(img.data[0, 0, 0, :, :]))
ax[0, 1].set_title('Iterative SENSE', fontsize=10)
ax[0, 2].imshow(torch.abs(img_manual[0, 0, 0, :, :]))
ax[0, 2].set_title('"Manual" Iterative SENSE', fontsize=10)
```

not be clearer for users what is a tensor and what is a MRpro object?

Reply:
rss of a single-coil image is the magnitude of the image...

so I would always do img_direct.rss() instead of img_direct.data.abs() (which, in turn, I would prefer over torch.abs(img_direct.data) )

RSS is a function of IData exactly for plotting..

Reply:
and IData always has the coil dimension....